sklearn/linear_model/_base.py

```diff
@@ -790,15 +793,17 @@ def _check_precomputed_gram_matrix(
     expected = np.dot(v1, v2)
     actual = precompute[f1, f2]
 
-    if not np.isclose(expected, actual, rtol=rtol, atol=atol):
-        raise ValueError(
-            "Gram matrix passed in via 'precompute' parameter "
-            "did not pass validation when a single element was "
-            "checked - please check that it was computed "
-            f"properly. For element ({f1},{f2}) we computed "
-            f"{expected} but the user-supplied value was "
-            f"{actual}."
-        )
+    err_msg = (
+        "Gram matrix passed in via 'precompute' parameter "
+        "did not pass validation when a single element was "
+        "checked - please check that it was computed "
+        f"properly. For element ({f1},{f2}) we computed "
+        f"{expected} but the user-supplied value was "
+        f"{actual}."
+    )
+    assert_allclose(
```

Comment:
This changes the exception class to `ValueError`. Also importing from `_testing` in non-testing code has the consequence of importing `pytest` (if `pytest` is installed). Loading an unused module can increase the startup time of programs.

I am in favor of copying the `rtol` selection logic here.